components/camel-jetty9/src/main/java/org/apache/camel/component/jetty9/JettyContentExchange9.java

```diff
@@ -214,14 +220,28 @@ public void onFailure(Request request, Throwable failure) {
             }
 
         };
-        BufferingResponseListener responseListener = new BufferingResponseListener() {
+
+        InputStreamResponseListener responseListener = new InputStreamResponseListener() {
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+            @Override
+            public void onContent(Response response, ByteBuffer content, Callback callback) {
+                if (response.getStatus() != SC_OK) {
```

Comment:
How do you know that the response is finished? What if there is a lot of data, wont this onContent be called multiple times?